projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -354,6 +362,49 @@ def is_vectorizer_disabled_query(self) -> sql.Composed:
             self.vectorizer_table_ident,
         )
 
+    @cached_property
+    def requeue_or_remove_work_query(self) -> sql.Composed:
+        return sql.SQL("""
+            MERGE INTO {queue_table} AS target
+            USING (
+                SELECT
+                    {pk_fields}
+                FROM
+                    {queue_table}
+                WHERE
+                    ({pk_fields}) IN ({pk_values})
+                AND
+                    (loading_retry_after is null or loading_retry_after < now())
+            ) AS source
+            ON {merge_predicates}
+            WHEN MATCHED
+                AND target.loading_retries >= %(loading_retries)s THEN
+                    DELETE
```

Comment:
This makes me unhappy as this leaves no record of what permanently failed.
I see a few options
- add a tombstone column
- add a tombstone table
- just increment loading_retries again and have the "get stuff from work queue" logic ignore items that have to high a retry count.